### PR TITLE
CY-2960 Widgets linter fixes

### PR DESCRIPTION
**Pull request by kubama**

### CODE REVIEW

File: widgets/blueprints/src/props/DataPropType.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export default PropTypes.shape({ items: PropTypes.arrayOf(PropTypes.shape({})), total: PropTypes.number });
```
Comment (qooban) [inline widgets/blueprints/src/props/DataPropType.js:1]: Maybe worth to extend it a bit (`items` field) to be more descriptive?

File: widgets/blueprintCatalog/src/props/DataPropType.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export default PropTypes.shape({
+    isAuthenticated: PropTypes.bool,
+    items: PropTypes.arrayOf(PropTypes.shape({})),
```
Comment (qooban) [inline widgets/blueprintCatalog/src/props/DataPropType.js:3]: Probably we can be more descriptive here with `items`.

File: widgets/blueprintCatalog/src/UploadModal.js [hunk 3]
```diff
@@ -21,16 +23,19 @@ export default class UploadModal extends React.Component {
...
         open: PropTypes.bool.isRequired,
         onHide: PropTypes.func.isRequired,
-        toolbox: PropTypes.object.isRequired,
-        actions: PropTypes.object.isRequired,
-        defaultYamlFile: PropTypes.string
+        toolbox: Stage.Common.PropTypes.Toolbox.isRequired,
+        actions: ActionsPropType.isRequired,
+        defaultYamlFile: PropTypes.string,
+        zipUrl: PropTypes.string.isRequired,
+        imageUrl: PropTypes.string
```
Comment (qooban) [inline widgets/blueprintCatalog/src/UploadModal.js:33]: Let's sort it alphabetically.

I'm also putting required fields on the top, but I'm OK with just sorted.

File: widgets/blueprintInfo/src/BlueprintInfo.js [hunk 4]
```diff
@@ -97,3 +97,8 @@ export default class BlueprintInfo extends React.Component {
         );
     }
 }
+
+BlueprintInfo.propTypes = {
+    widget: Stage.Common.PropTypes.Widget.isRequired,
+    data: PropTypes.shape({}).isRequired
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/blueprintInfo/src/BlueprintInfo.js]: Done

File: widgets/blueprintUploadButton/src/UploadButton.js [hunk 5]
```diff
@@ -42,3 +42,7 @@ export default class extends React.Component {
         );
     }
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/blueprintUploadButton/src/UploadButton.js:42]: Done

File: widgets/common/src/GroupState.js [hunk 6]
```diff
@@ -17,7 +17,7 @@ class GroupState extends React.Component {
             icon: PropTypes.string.isRequired
         }).isRequired,
         className: PropTypes.string.isRequired,
-        description: PropTypes.object.isRequired
+        description: PropTypes.shape({}).isRequired
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/common/src/GroupState.js]: Done

File: widgets/common/src/RolesPresenter.js [hunk 7]
```diff
@@ -8,7 +8,7 @@ export default class RolesPresenter extends React.Component {
     }
 
     static propTypes = {
-        groupRoles: PropTypes.object,
+        groupRoles: PropTypes.shape({}),
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/common/src/RolesPresenter.js]: > 1. Maybe we can make it functional component by moving out static the same way as `defaultProps` are defined?

Done, but it's not necessary to convert to functional component to achieve this.

> 2. If `groupRoles` default value is `[]` probably we should have different type here, right?

Actually it should be the other way around - I fixed the default value.

File: widgets/executions/src/tasksGraph/GraphNode.js [hunk 8]
```diff
@@ -123,6 +123,8 @@ GraphNode.propTypes = {
     graphNode: PropTypes.shape({
         labels: PropTypes.arrayOf(
             PropTypes.shape({
+                display_text: PropTypes.arrayOf(PropTypes.string),
+                display_title: PropTypes.arrayOf(PropTypes.string),
                 displayTitle: PropTypes.arrayOf(PropTypes.string),
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/executions/src/tasksGraph/GraphNode.js]: It seems that  `displayTitle` is not used - removed

File: widgets/executions/src/ExecutionsTable.js [hunk 9]
```diff
@@ -401,3 +401,14 @@ export default class ExecutionsTable extends React.Component {
         );
     }
 }
+
+ExecutionsTable.propTypes = {
+    data: PropTypes.shape({
+        blueprintId: PropTypes.bool,
+        deploymentId: PropTypes.bool,
+        items: PropTypes.arrayOf(PropTypes.shape({})),
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/executions/src/ExecutionsTable.js]: Done

File: widgets/events/src/EventsTable.js [hunk 10]
```diff
@@ -271,3 +271,16 @@ export default class EventsTable extends React.Component {
         );
     }
 }
+
+EventsTable.propTypes = {
+    data: PropTypes.shape({
+        blueprintId: PropTypes.arrayOf(PropTypes.string),
+        deploymentId: PropTypes.arrayOf(PropTypes.string),
+        executionId: PropTypes.arrayOf(PropTypes.string),
+        items: PropTypes.arrayOf(PropTypes.shape({})),
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/events/src/EventsTable.js]: Done

File: widgets/deployments/src/props/DeploymentsViewPropTypes.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
+export default {
+    data: PropTypes.shape({
+        blueprintId: PropTypes.string,
+        items: PropTypes.arrayOf(PropTypes.shape({})),
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/deployments/src/props/DeploymentsViewPropTypes.js]: Done

File: widgets/deploymentWizardButtons/src/wizard/WizardModal.js [hunk 12]
```diff
@@ -390,3 +369,27 @@ export default class WizardModal extends React.Component {
...
+        id: PropTypes.string.isRequired,
+        title: PropTypes.string.isRequired,
+        description: PropTypes.string.isRequired,
+        Content: PropTypes.func.isRequired,
+        Actions: PropTypes.func.isRequired
+    })
+);
+
+/**
+ * @property {boolean} open Controls whether or not the wizard modal is displayed
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/deploymentWizardButtons/src/wizard/WizardModal.js:385]: Done

File: widgets/deploymentWizardButtons/src/steps/SecretsStep.js [hunk 13]
```diff
@@ -103,16 +103,14 @@ class SecretsStepContent extends React.Component {
                     }
                 }
 
-                return { stepData, secretsInManager };
+                return stepData;
             })
             .then(
-                ({ stepData, secretsInManager }) =>
-                    new Promise(resolve =>
-                        this.setState({ secretsInManager }, () => {
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/deploymentWizardButtons/src/steps/SecretsStep.js:111]: Done

File: widgets/common/src/props/Widget.js [hunk 14]
```diff
@@ -0,0 +1 @@
+export default PropTypes.shape({ configuration: PropTypes.shape({}) });
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/common/src/props/Widget.js]: Done

File: widgets/common/src/WorkflowsMenu.js [hunk 15]
```diff
@@ -89,9 +89,9 @@ class AccordionWorkflowsMenu extends React.Component {
 
 class WorkflowsMenu extends React.Component {
     static propTypes = {
-        workflows: PropTypes.array.isRequired,
+        workflows: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/common/src/WorkflowsMenu.js]: Done

File: widgets/common/src/WorkflowsMenu.js [hunk 16]
```diff
@@ -4,7 +4,7 @@
 
 class WorkflowsMenuItems extends React.Component {
     static propTypes = {
-        workflows: PropTypes.array.isRequired,
+        workflows: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/common/src/WorkflowsMenu.js]: Done

File: widgets/blueprintCatalog/src/UploadModal.js (hunk 3 above)
Comment (kubama) [reply widgets/blueprintCatalog/src/UploadModal.js]: Done

File: widgets/blueprints/src/props/DataPropType.js (hunk 1 above)
Comment (kubama) [reply widgets/blueprints/src/props/DataPropType.js]: Done

File: widgets/executions/src/tasksGraph/ExecutionWorkflowGraph.js [hunk 17]
```diff
@@ -19,9 +19,10 @@ export default class ExecutionWorkflowGraph extends React.Component {
      * @property {Any} [selectedExecution] - Used to pull the execution's tasks graphs and corresponding operations' lists
      */
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/executions/src/tasksGraph/ExecutionWorkflowGraph.js]: Done

**Pull request merged**